packages/components/CONTRIBUTING.md

```diff
@@ -236,6 +232,74 @@ A couple of good examples of how hooks are used for composition are:
 
 TDB -->
 
+## Naming Conventions
+
+It is recommended for compound components to use dot notation to separate the namespace from the individual component names. The top-level compound component should be called as the namespace (no dot notation).
+
+Dedicated React context should also use the dot notation, while hooks should not.
+
+```tsx
+// Component.tsx
+//=======================
+import { forwardRef, createContext } from '@wordpress/element';
+
+function TopLevelComponent = forwardRef( function TopLevel( props, ref ) { /* ... */ });
```

Comment:
Should we have the unforwarded be declared separately as `UnforwardedTopLevelComponent` as we do for our components usually?